libsentrykube/iap.py

```diff
@@ -66,59 +88,63 @@ def ensure_iap_tunnel(ctx: click.core.Context, quiet: bool = False) -> str:
             )
 
         _dns_check()
-        control_plane_ip = urlparse(cluster["cluster"]["server"]).hostname
-        cluster["cluster"]["server"] = f"https://kubernetes:{port}"
+        control_plane_host = urlparse(cluster["cluster"]["server"]).hostname
+        use_dns_endpoint = _dns_endpoint_check(control_plane_host, quiet)
+
+        if not use_dns_endpoint:
+            cluster["cluster"]["server"] = f"https://kubernetes:{port}"
 
         tmp_kubeconfig_path = os.path.join(
             os.path.dirname(KUBE_CONFIG_PATH), f"sentry-kube.config.{port}.yaml"
         )
         with open(tmp_kubeconfig_path, "w") as tmp_cf:
             yaml.dump(kubeconfig, tmp_cf)
 
-    port_fwd = f"{port}:{control_plane_ip}:443"
-
-    if not _tcp_port_check(port):
-        if not quiet:
-            click.echo(f"Spawning port forwarding for {port_fwd}")
-
-        subprocess.Popen(
-            build_ssh_command(
-                ctx,
-                host=KUBECTL_JUMP_HOST,
-                project=None,
-                user=None,
-                ssh_key_file=None,
-                region=None,
-                zone=None,
-                # -N -- Do not execute remote command
-                # -T -- do not allocate tty
-                # -f -- go to background, before the command execution
-                # ExitOnForwardFailure=yes
-                # Terminate the connection if it cannot set up all requested dynamic,
-                # tunnel, local, and remote port forwardings.
-                ssh_args=(
-                    "-NTf",
-                    "-o",
-                    "ExitOnForwardFailure=yes",
-                    "-L",
-                    port_fwd,
-                ),
-            ),
-            # spawn as detached process
-            start_new_session=True,
-            stdout=subprocess.DEVNULL,
-            stderr=subprocess.STDOUT,
-        )
-        # try port_check_attempts times if the port forwarding is in place
-        port_check_attempts = 10
-        for _ in range(port_check_attempts):
+    # Skip all of this junk if we're using the DNS endpoint
+    if not use_dns_endpoint:
```

Comment:
nit: please invert the if block and return early so the whole block below is less indented and more readable.